src/controllers/controllerpresetfilehandler.cpp

```diff
@@ -13,56 +13,53 @@
 #include "controllers/hid/hidcontrollerpresetfilehandler.h"
 
 // static
-ControllerPresetPointer ControllerPresetFileHandler::loadPreset(const QString& pathOrFilename,
-                                                                const QStringList& presetPaths) {
-    qDebug() << "Searching for controller preset" << pathOrFilename
-             << "in paths:" << presetPaths.join(",");
-    QString scriptPath = ControllerManager::getAbsolutePath(pathOrFilename,
-                                                            presetPaths);
-
-    if (scriptPath.isEmpty()) {
-        qDebug() << "Could not find" << pathOrFilename
-                 << "in any preset path.";
+ControllerPresetPointer ControllerPresetFileHandler::loadPreset(
+        const QFileInfo& presetFile, const QDir& systemPresetsPath) {
+    VERIFY_OR_DEBUG_ASSERT(presetFile.exists() && presetFile.isReadable()) {
```

Comment:
I don't think this should be a debug assertion. This function is taking in untrusted data from the filesystem that has not been validated yet.

Reply:
Correct. I was originally planning to check this beforehand, but that would still suffer from TOCTOU issues...